### PR TITLE
[EASI-3000] Read View Users Contribute Discussions

### PR DESCRIPTION
**Pull request by OddTomBrooks**
# EASI-3000

## Changes and Description

- Read-only users can create a discussion
- Users can select a role when creating a discussion
- The `mostRecentDiscussionRoleSelection` returns the most recent role selection for a user

## How to test this change

- Run `scripts/dev down && scripts/dev up:backend && scripts/dev db:seed`
  - Assert that a user_role column has been added to the `plan_discussion` table
- Initialize model plan variables using standard endpoints
- Create a user discussion
- Run the `mostRecentDiscussionRoleSelection` endpoint
  - Assert that the most recent user role selection matches with the most recent created or modified entry in the database

## PR Author Review Checklist

- [x] Met the ticket's acceptance criteria, or will meet them in a subsequent PR.
- [ ] Added or updated tests for backend resolvers or other functions as needed.
- [ ] Added or updated client tests for new components, parent components, functions, or e2e tests as necessary.
- [x] Updated the [Postman Collection](../MINT.postman_collection.json) if necessary.

## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.

### CODE REVIEW

**Review by ClayBenson94 — CHANGES_REQUESTED** [review]

File: migrations/V95__Update_Model_Plan_Abbreviation.sql
[the remark this answers is not in the document]
Comment (OddTomBrooks) [reply migrations/V95__Update_Model_Plan_Abbreviation.sql:1]: This feature branch was created off of the previous feature branch to reduce merge work. You are correct that this file should be reverted to the main revision and whitespace changes ignored.

File: pkg/storage/plan_discussionStore.go [hunk 1]
```diff
@@ -242,3 +245,22 @@ func (s *Store) DiscussionReplyByID(logger *zap.Logger, id uuid.UUID) (*models.D
 
 	return discussionReply, nil
 }
+
+// GetMostRecentDiscussionRoleSelection retrieves the latest role selection for a given user
+func (s *Store) GetMostRecentDiscussionRoleSelection(logger *zap.Logger, userID uuid.UUID) (models.DiscussionUserRole, error) {
+	print("GetLatestDiscussionRoleSelection test\n\n")
```
[the remark this answers is not in the document]
Comment (OddTomBrooks) [reply pkg/storage/plan_discussionStore.go:251]: Yep! Ghost debug code. Good catch.

**Review by ClayBenson94 — DISMISSED** [review]

**Review by patrickseguraoddball — DISMISSED** [review]
I think there needs to be userRole and useRoleDescription on replies as well.

**Comment by OddTomBrooks** [comment]
> I think there needs to be userRole and useRoleDescription on replies as well.

Ok. On it.

**Comment by patrickseguraoddball** [comment]
Is `mostRecentDiscussionRoleSelection` return value something we could have on the modelPlan itself?  Having to make a separate query each time is doable, but this would alleviate a lot of overhead for the FE.  If it was a field on the model plan that we could query would be awesome

**Comment by OddTomBrooks** [comment]
> Is `mostRecentDiscussionRoleSelection` return value something we could have on the modelPlan itself? Having to make a separate query each time is doable, but this would alleviate a lot of overhead for the FE. If it was a field on the model plan that we could query when we fetch the modelPlan/discussions would be awesome

It's doable. This introduces a pseudo-statefulness which is probably fine, but I'd like to discuss it with @ClayBenson94 tomorrow to make sure that's something we'd like to do from an architectural standpoint. As it stands, the most recent discussion role selection is a value calculated from a database query.

The total lift on this is fairly minor, so if Clay is fine with it I'll make the change tomorrow.

**Comment by OddTomBrooks** [comment]
Smoke testing works as expected. No issues noticed.

**Review by ClayBenson94 — APPROVED** [review]

**Pull request merged**